src/Services/Core/MemberService.php

```diff
@@ -233,4 +246,105 @@ public function getLatestBoard(): array
         return $boardMembers;
 
     }
+
+    /**
+     * @param array $fields
+     * @return Member
+     * @throws \Exception
+     */
+    public function createPhoto(int $id, ?array $fields): String
+    {
+        if ($fields['file'] == null) {
+            $msg = 'File is empty in given parameters';
+            $this->logger->error($msg);
+            throw new KerosException($msg, 400);
+        }
+        $file = $fields['file'];
+
+        $id = Validator::requiredId($id);
+        $member = $this->getOne($id);
+
+        if (!$member) {
+            throw new KerosException("The member could not be found", 404);
+        }
+
+        $filename = $member->getProfilePicture();
+
+        if ($filename) {
+            $filepath =  $this->kerosConfig['MEMBER_PHOTO_DIRECTORY'] . $filename;
+            if (file_exists($filepath)){
+                unlink($filepath);
+            }
+        }
+        
+        $filename = $this->directoryManager->uniqueFilename($file, false, '');

```

Comment:
`uniqueFilename` permet d'avoir un nom de fichier unique pour la `location` (cad path) donné en 3e attribut. En ne mettant rien, je pense que ça va chercher un nom unique dans le dossier où se trouve le fichier. Il faudrait donc remplacer par `$filename = $this->directoryManager->uniqueFilename($file, false, $this->kerosConfig['MEMBER_PHOTO_DIRECTORY']);` pour avoir un nom unique dans le dossier des photos des membres.